cartesi-rollups_versioned_docs/version-1.5/rollups-apis/graphql/basics.md

```diff
@@ -3,89 +3,92 @@ id: overview
 
-This specification is displayed in a more accessible and navigable way in the next sections.
 
+Some key features of the GraphQL API include the ability to:
 
-## Queries
+- Retrieve detailed information about individual inputs, notices, vouchers, and reports.
+- List and paginate through collections of these entities.
+- Access proof data for notices and vouchers, allowing for validation on the base layer.
+- Filter and sort results based on various criteria.
 
-A number of top-level queries are available in order to retrieve rollup information for a Cartesi dApp.
+## Basic Query Structure
 
-In GraphQL, submitting a query involves defining parameters for filtering the entries to retrieve, and also specifying the data fields of interest, which can span any objects linked to the entry being retrieved.
+GraphQL queries in the Cartesi Rollups API typically involve specifying:
 
-For example, the following query retrieves the number of the base layer block in which the [input](./objects/input.mdx) was recorded:
+1. The type of entity you're querying (input, notice, voucher, or report).
+2. Any necessary arguments (e.g., index values, pagination parameters).
+3. The fields you want to retrieve from the entity.
 
-```
-input(index: "1") {
-    blockNumber
-  }
-```
-
-You can submit the query above as an HTTP POST request, specifying the Content-Type as `application/json`. For instance, using `curl` you could submit it as follows:
-
-```
-curl 'https://<graphql_url>' -H 'Content-Type: application/json' -d '{"query":"{ input(index:1) {blockNumber} }"}'
-```
+Here's a basic example of querying an input:
 
-The response of which would be something like this:
-
-```
-{
-  "data": {
-    "input": {
-      "blockNumber": 8629116
-    }
+```graphql
+query {
+  input(index: 1) {
+    index
+    status
+    timestamp
+    msgSender
+    payload
   }
 }
 ```
 
-You can also retrieve linked information from the input. For example, the following query would retrieve notices from this particular input with support for pagination and with total number of entries that match the query:
+## Pagination
 
-```
-input(index: "1") {
-    notices {
-      totalCount
-    }
-  }
-```
+The API uses cursor-based pagination for queries that return collections of entities (e.g., listing all notices). This is implemented through connection types (e.g.,[`InputConnection`](./objects/input-connection.md), [`NoticeConnection`](./objects/notice-connection.md)) that include `edges` and `pageInfo` fields.
 
-The response of which would be something like this:
+Example of a paginated query:
 
-```
-{
-  "data": {
-    "input": {
-      "notices": {
-        "totalCount": 6
+```graphql
+query {
+  notices(first: 5, after: "cursor_value") {
+    edges {
+      node {
+        index
+        payload
       }
+      cursor
+    }
+    pageInfo {
+      hasNextPage
+      endCursor
     }
   }
 }
 ```
 
-## Response format
+## Proof Data
```

Comment:
I'm not sure how Proof Data is helping in this page flow. We're talking about basic graphQL querying.

Reply:
Yeah, we could move it. 

Looking at the sub-menu structure, I would also prefer to have Notices and Vouchers closer, meaning either Reports come first or they come last. 

![image](https://github.com/user-attachments/assets/ab233a01-ae05-442b-8e3d-f1fc7097862d)


Then we have two options, put a small section under Notices for 'Proof data' or create a subsection just for it.

Reply:
Resolved